### ISSUE TITLE
The Dot generator does not escape backslashes, causes rendering issues

### ISSUE THREAD

**Issue by sbstp** [bluss/petgraph#214]
The issue can be seen inside the Regex object in these graphs (nodes and edges).

Here's the unedited version:

```dot
digraph {
    0 [label="PatternInfo { pattern: DontMatch, group: None }"]
    1 [label="PatternInfo { pattern: Regex(\w+), group: Some(\"series\") }"]
    2 [label="PatternInfo { pattern: Regex(^s\d\d?e\d\d?$), group: Some(\"season_episode\") }"]
    3 [label="PatternInfo { pattern: Regex(s\d\d?), group: Some(\"season\") }"]
    4 [label="PatternInfo { pattern: Regex(e\d\d?), group: Some(\"episode\") }"]
    5 [label="PatternInfo { pattern: Regex(\d\d?), group: Some(\"season\") }"]
    6 [label="PatternInfo { pattern: Regex(\d\d?), group: Some(\"episode\") }"]
    0 -> 1 [label="PatternInfo { pattern: Regex(\w+), group: Some(\"series\") }"]
    1 -> 1 [label="PatternInfo { pattern: Regex(\w+), group: Some(\"series\") }"]
    1 -> 2 [label="PatternInfo { pattern: Regex(^s\d\d?e\d\d?$), group: Some(\"season_episode\") }"]
    1 -> 3 [label="PatternInfo { pattern: Regex(s\d\d?), group: Some(\"season\") }"]
    3 -> 4 [label="PatternInfo { pattern: Regex(e\d\d?), group: Some(\"episode\") }"]
    1 -> 5 [label="PatternInfo { pattern: Regex(\d\d?), group: Some(\"season\") }"]
    5 -> 6 [label="PatternInfo { pattern: Regex(\d\d?), group: Some(\"episode\") }"]
}
```
And the image generated with this: https://i.imgur.com/hX8W6JK.png

Here's a version that I have manually escaped
```dot
digraph {
    0 [label="PatternInfo { pattern: DontMatch, group: None }"]
    1 [label="PatternInfo { pattern: Regex(\\w+), group: Some(\"series\") }"]
    2 [label="PatternInfo { pattern: Regex(^s\\d\\d?e\\d\\d?$), group: Some(\"season_episode\") }"]
    3 [label="PatternInfo { pattern: Regex(s\\d\\d?), group: Some(\"season\") }"]
    4 [label="PatternInfo { pattern: Regex(e\\d\\d?), group: Some(\"episode\") }"]
    5 [label="PatternInfo { pattern: Regex(\\d\\d?), group: Some(\"season\") }"]
    6 [label="PatternInfo { pattern: Regex(\\d\\d?), group: Some(\"episode\") }"]
    0 -> 1 [label="PatternInfo { pattern: Regex(\\w+), group: Some(\"series\") }"]
    1 -> 1 [label="PatternInfo { pattern: Regex(\\w+), group: Some(\"series\") }"]
    1 -> 2 [label="PatternInfo { pattern: Regex(^s\\d\\d?e\\d\\d?$), group: Some(\"season_episode\") }"]
    1 -> 3 [label="PatternInfo { pattern: Regex(s\\d\\d?), group: Some(\"season\") }"]
    3 -> 4 [label="PatternInfo { pattern: Regex(e\\d\\d?), group: Some(\"episode\") }"]
    1 -> 5 [label="PatternInfo { pattern: Regex(\\d\\d?), group: Some(\"season\") }"]
    5 -> 6 [label="PatternInfo { pattern: Regex(\\d\\d?), group: Some(\"episode\") }"]
}
````
The image generated with this: https://i.imgur.com/j9WhF4L.png

I can try to fix this issue, shouldn't be too hard. Might look into other things that need escaping as well.

**Comment by bluss** [comment]
Sounds good. There's an escaper, so it should only need to be patched https://github.com/bluss/petgraph/blob/master/src/dot.rs#L184 (?)

**Comment by bluss** [comment]
Fixed in #215
